### PR TITLE
ft(S3C-3041): Add v2 Toggle based on environment variable

### PR DESCRIPTION
**Pull request by tmacro**

### CODE REVIEW

**Review by anurag4DSB — APPROVED** [review]

**Comment by miniscruff** [comment]
So this is a 2.0 toggle within v7/v8?

**Comment by tmacro** [comment]
Its a v2 in the sense that its a complete rewrite and rearchitecture of Utapi. The project versioning is not semantic and is tied to RING releases. For non technical reasons they reside in the same git repo.

**Comment by miniscruff** [comment]
> Its a v2 in the sense that its a complete rewrite and rearchitecture of Utapi. The project versioning is not semantic and is tied to RING releases. For non technical reasons they reside in the same git repo.

hmm ok, I mean what you said makes sense but at the same time doesn't make sense... lol

**Review by miniscruff — APPROVED** [review]
LGTM

File: tests/unit/testV2Shim.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/* eslint-disable no-console */
+const { fork } = require('child_process');
+const path = require('path');
+const assert = require('assert');
+
+const checker = path.resolve(__dirname, '../utils/exportCheck');
+
+function execShim(version, env, callback) {
```
Comment (naren-scality) [inline tests/unit/testV2Shim.js:8]: Do you need `version` here? Can this not be decided using `process.env.ENABLE_UTAPI_V2`?

File: tests/utils/exportCheck.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    UtapiClient: require('../../lib/UtapiClient'),
+    UtapiReplay: require('../../lib/UtapiReplay'),
+    UtapiReindex: require('../../lib/UtapiReindex'),
+};
+
+const version2Exports = {
+    version: 2,
+};
+
+const versionToCheck = process.argv[2];
```
Comment (naren-scality) [inline tests/utils/exportCheck.js:18]: Same question as my other comment on `version` and `process.env.ENABLE_UTAPI_V2`!

**Review by naren-rajendran — APPROVED** [review]

**Pull request merged**